docs/deletion-propagation.md

```diff
@@ -0,0 +1,112 @@
+# Deletion Propagation
+
+Historically, Config Sync has handled applying and pruning resource objects
+based on their presence or absense from a source of truth (Git, OCI, Helm).
+
+In Config Sync v1.15.0, a new feature was added to delete all previously applied
+objects, even if they still exist in the source of truth. This feature is
+called Deletion Propagation, because it allows you to enable the feature on a
+Sync object (RootSync or RepoSync) and delete the Sync object, then Config Sync
+will automatically propagate that deletion to every object managed by that Sync
+object, according to its current inventory (ResourceGroup object).
+
+This feature is useful in a number of cases, like migrating to a new namespace
+or cluster, cleaning up after a demo or experiment, uninstalling an application
+or other software package, etc.
+
+## Usage
+
+Deletion Propagation is disabled by default, when the annotation is not present.
+
+To enable Deletion Propagation, set the following annotation on the RootSync or
+RepoSync object:
+
+```yaml
+configsync.gke.io/deletion-propagation-policy: Foreground
+```
+
+To disable Deletion Propagation, either remove the annotation or explicitly
+change the value to `Orphan`:
+
+```yaml
+configsync.gke.io/deletion-propagation-policy: Orphan
+```
+
+## Example
+
+To delete all the objects managed by the RootSync named `example`, first patch
+the RootSync to enable Deletion Propagation:
+
+```bash
+kubectl patch RootSync example --namespace config-manaement-system --type merge \
+  --patch '{"metadata":{"annotations":{"configsync.gke.io/deletion-propagation-policy":"Foreground"}}}'
+```
+
+Then delete the RootSync and wait for it to be deleted:
+
+```bash
+kubectl delete RootSync example --namespace config-manaement-system --wait
+```
+
+## Implimentation Details
```

Comment:
`Implementation` (also fix spelling below)

Reply:
finally installed a spellchecker in vscode and fixed several spell checks.